includes/libs/tests/JavaScriptMinifierTest.php

```diff
@@ -0,0 +1,212 @@
+			[ "function x(){}/  x/g", "function x(){}/  x/g" ],
+			[ "+function x(){}/  x/g", "+function x(){}/x/g" ],
+
+			// Multiline quoted string
+			[ "var foo=\"\\\nblah\\\n\";", "var foo=\"\\\nblah\\\n\";" ],
+
+			// Multiline quoted string followed by string with spaces
+			[
+				"var foo=\"\\\nblah\\\n\";\nvar baz = \" foo \";\n",
+				"var foo=\"\\\nblah\\\n\";var baz=\" foo \";"
+			],
+
+			// URL in quoted string ( // is not a comment)
+			[
+				"aNode.setAttribute('href','http://foo.bar.org/baz');",
+				"aNode.setAttribute('href','http://foo.bar.org/baz');"
+			],
+
+			// URL in quoted string after multiline quoted string
+			[
+				"var foo=\"\\\nblah\\\n\";\naNode.setAttribute('href','http://foo.bar.org/baz');",
+				"var foo=\"\\\nblah\\\n\";aNode.setAttribute('href','http://foo.bar.org/baz');"
+			],
+
+			// Division vs. regex nastiness
+			[
+				"alert( (10+10) / '/'.charCodeAt( 0 ) + '//' );",
+				"alert((10+10)/'/'.charCodeAt(0)+'//');"
+			],
+			[ "if(1)/a /g.exec('Pa ss');", "if(1)/a /g.exec('Pa ss');" ],
+
+			// newline insertion after 1000 chars: break after the "++", not before
+			[ str_repeat( ';', 996 ) . "if(x++);", str_repeat( ';', 996 ) . "if(x++\n);" ],
+
+			// Unicode letter characters should pass through ok in identifiers (T33187)
+			[ "var KaŝSkatolVal = {}", 'var KaŝSkatolVal={}' ],
+
+			// Per spec unicode char escape values should work in identifiers,
+			// as long as it's a valid char. In future it might get normalized.
+			[ "var Ka\\u015dSkatolVal = {}", 'var Ka\\u015dSkatolVal={}' ],
+
+			// Some structures that might look invalid at first sight
+			[ "var a = 5.;", "var a=5.;" ],
+			[ "5.0.toString();", "5.0.toString();" ],
+			[ "5..toString();", "5..toString();" ],
+			[ "5...toString();", false ],
+			[ "5.\n.toString();", '5..toString();' ],
+
+			// Boolean minification (!0 / !1)
+			[ "var a = { b: true };", "var a={b:!0};" ],
+			[ "var a = { true: 12 };", "var a={true:12};", false ],
+			[ "a.true = 12;", "a.true=12;", false ],
+			[ "a.foo = true;", "a.foo=!0;" ],
+			[ "a.foo = false;", "a.foo=!1;" ],
+		];
+	}
+
+	/**
+	 * @dataProvider provideCases
+	 * @covers JavaScriptMinifier::minify
+	 */
+	public function testJavaScriptMinifierOutput( $code, $expectedOutput, $expectedValid = true ) {
+		$minified = JavaScriptMinifier::minify( $code );
+
+		// JSMin+'s parser will throw an exception if output is not valid JS.
+		// suppression of warnings needed for stupid crap
+		if ( $expectedValid ) {
+			wfSuppressWarnings();
+			$parser = new JSParser();
+			wfRestoreWarnings();
+			$parser->parse( $minified, 'minify-test.js', 1 );
+		}
+
+		$this->assertEquals(
+			$expectedOutput,
+			$minified,
+			"Minified output should be in the form expected."
+		);
+	}
+
+	public static function provideExponentLineBreaking() {
+		return [
+			[
+				// This one gets interpreted all together by the prior code;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this commented out code. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS125)